packages/git/src/common/git-model.ts

```diff
@@ -110,6 +110,15 @@ export interface Repository {
 
 }
 
+export namespace Repository {
+    export function equal(repository: Repository | undefined, repository2: Repository | undefined): boolean {
+        if (repository && repository2) {
+            return repository.localUri === repository2.localUri;
+        }
+        return repository === repository2;
```

Comment:
That is `undefined === undefined`. Maybe the first arg doesn't need to be optional?

Reply:
there is a client which can pass both `undefined`: https://github.com/theia-ide/theia/blob/a8389fa95a6132ef6af13c0c0f7de9f5846ca05b/packages/git/src/browser/git-repository-provider.ts#L73